cli/cmd/generate_azure.go

```diff
@@ -0,0 +1,696 @@
+			Required: true,
+			Response: &config.ManagementGroupId,
+		},
+	}); err != nil {
+		return err
+	}
+	return nil
+}
+
+func promptAzureAdIntegrationQuestions(config *azure.GenerateAzureTfConfigurationArgs) error {
+
+	if err := SurveyMultipleQuestionWithValidation([]SurveyQuestionWithValidationArgs{
+		{
+			Prompt:   &survey.Input{Message: QuestionADApplicationPass, Default: config.AdApplicationPassword},
+			Required: true,
+			Response: &config.AdApplicationPassword,
+		},
+		{
+			Prompt:   &survey.Input{Message: QuestionADApplicationId, Default: config.AdApplicationId},
+			Required: true,
+			Response: &config.AdApplicationId,
+		},
+		{
+			Prompt:   &survey.Input{Message: QuestionADServicePrincpleId, Default: config.AdServicePrincipalId},
+			Required: true,
+			Response: &config.AdServicePrincipalId,
+		},
+	}); err != nil {
+		return err
+	}
+	return nil
+}
+
+func promptCustomizeAzureOutputLocation(extraState *AzureGenerateCommandExtraState) error {
+
+	if err := SurveyMultipleQuestionWithValidation([]SurveyQuestionWithValidationArgs{
+		{
+			Prompt:   &survey.Input{Message: QuestionAzureCustomizeOutputLocation, Default: extraState.Output},
+			Response: &extraState.Output,
+		},
+	}); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func promptCustomizeAzureLoggingRegion(config *azure.GenerateAzureTfConfigurationArgs) error {
+	var region string
+	if err := SurveyMultipleQuestionWithValidation([]SurveyQuestionWithValidationArgs{
+		{
+			Prompt:   &survey.Input{Message: QuestionStorageLocation, Default: config.StorageLocation},
+			Response: &region,
+		},
+	}); err != nil {
+		return err
+	}
+	if err := validateStorageLocation(region); err != nil {
+		return err
+	}
+	config.StorageLocation = region
+	return nil
+}
+
+func askAdvancedAzureOptions(config *azure.GenerateAzureTfConfigurationArgs, extraState *AzureGenerateCommandExtraState) error {
+	answer := ""
+
+	// Prompt for options
+	for answer != AzureAdvancedOptDone {
+
+		// Set the initial options
+		options := []string{AzureUserIntegrationNames, AzureExistingStorageAcount, AzureSubscriptions, AzureManagmentGroup}
+		// Only ask about Active Directory information if one was requested to be created
+		if !config.CreateAdIntegration {
+			options = append(options, AdvancedAdIntegration)
+		}
+		options = append(options, AzureAdvancedOptLocation, AzureRegionStorage, AzureAdvancedOptDone)
+		if err := SurveyQuestionInteractiveOnly(SurveyQuestionWithValidationArgs{
+			Prompt: &survey.Select{
+				Message: "Which options would you like to enable?",
+				Options: options,
+			},
+			Response: &answer,
+		}); err != nil {
+			return err
+		}
+
+		// Based on response, prompt for actions
+		switch answer {
+		case AzureUserIntegrationNames:
+			if err := promptAzureIntegrationNameQuestions(config); err != nil {
+				return err
+			}
+		case AzureExistingStorageAcount:
+			if err := promptAzureStorageAccountQuestions(config); err != nil {
+				return err
+			}
```

Comment:
Just noticed that my GCP implementation suffers from this issue too.
We should probably only display this option if the user is has answered yes to performing an Activity log integration

Reply:
Perhaps the best we can do is to ensure that all questions in `promptAzureStorageAccountQuestions` have a check on `&config.ActivityLog`

Reply:
Yes storage account input is only required when an activity log integration is enabled and also the management information is only applicable in the case where a config integration is enabled, should put that logic in so that the these options are only shown in the case where that is the case

Reply:
Was this update made? I am still able to configure storage account settings when I only set config integration to true

Reply:
Did not make it into the version you looked at, updated now